src/neptune/new/internal/operation_processors/async_operation_processor.py

```diff
@@ -26,38 +28,51 @@
     Optional,
 )
 
+from neptune.new.constants import (
+    ASYNC_DIRECTORY,
+    NEPTUNE_DATA_DIRECTORY,
+)
 from neptune.new.exceptions import NeptuneSynchronizationAlreadyStoppedException
 from neptune.new.internal.backends.neptune_backend import NeptuneBackend
 from neptune.new.internal.container_type import ContainerType
 from neptune.new.internal.disk_queue import DiskQueue
 from neptune.new.internal.id_formats import UniqueId
 from neptune.new.internal.operation import Operation
 from neptune.new.internal.operation_processors.operation_processor import OperationProcessor
+from neptune.new.internal.operation_processors.operation_storage import OperationStorage
 from neptune.new.internal.threading.daemon import Daemon
 from neptune.new.internal.utils.logger import logger
+from neptune.new.sync.utils import create_dir_name
 
 # pylint: disable=protected-access
 
 _logger = logging.getLogger(__name__)
 
 
-class AsyncOperationProcessor(OperationProcessor):
+class AsyncOperationProcessor(OperationProcessor, OperationStorage):
```

Comment:
Same as we *have* a `DiskQueue` we should *have* a storage for operations instead of multi-inheritance (composition over inheritance)

Reply:
Good point, especially since we're going to move `OperationStorage` to `DiskQueue` in the future.